katz/src/main/kotlin/katz/effects/IO.kt

```diff
@@ -0,0 +1,220 @@
+package katz
+
+import katz.effects.internal.AndThen
+import katz.effects.internal.Platform
+import katz.effects.internal.Platform.unsafeResync
+
+sealed class IO<out A> {
+    abstract fun <B> map(f: (A) -> B): IO<B>
+
+    fun <B> flatMap(f: (A) -> IO<B>): IO<B> =
+            flatMapTotal(
+                    AndThen { a: A ->
+                        try {
+                            f(a)
+                        } catch (error: Throwable) {
```

Comment:
This diff is wrong, I made a mistake and is based off the IO branch. The IO changes are here https://github.com/FineCinnamon/Katz/pull/109

Could you please move the comments there?